CMakePCHCompiler.cmake

```diff
@@ -213,35 +242,54 @@ endmacro()
 # copies all compile definitions, flags and options to .pch subtarget
 function(__watch_pch_last_hook variable access value)
 	list(LENGTH CMAKE_PCH_COMPILER_TARGETS length)
-	foreach(index RANGE -${length} -1)
-		list(GET CMAKE_PCH_COMPILER_TARGETS ${index} target)
-		list(GET CMAKE_PCH_COMPILER_TARGET_FLAGS ${index} flags)
-		set(pch_target ${target}.pch)
-		foreach(property
-			COMPILE_DEFINITIONS
-			COMPILE_DEFINITIONS_DEBUG
-			COMPILE_DEFINITIONS_MINSIZEREL
-			COMPILE_DEFINITIONS_RELEASE
-			COMPILE_DEFINITIONS_RELWITHDEBINFO
-			COMPILE_FLAGS
-			COMPILE_OPTIONS
+	if(NOT 0 EQUAL length)			
+		foreach(index RANGE -${length} -1)
+			list(GET CMAKE_PCH_COMPILER_TARGETS ${index} target)			
+			set(pch_target ${target}.pch)
+			foreach(property
+				COMPILE_DEFINITIONS
+				COMPILE_DEFINITIONS_DEBUG
+				COMPILE_DEFINITIONS_MINSIZEREL
+				COMPILE_DEFINITIONS_RELEASE
+				COMPILE_DEFINITIONS_RELWITHDEBINFO
+				COMPILE_FLAGS
+				COMPILE_OPTIONS
+				INCLUDE_DIRECTORIES
+				# This list is likely incomplete.
+				# The commented out are done below.
+				#CXX_STANDARD
+				#POSITION_INDEPENDENT_CODE	
+				CXX_STANDARD_REQUIRED			
 			)
-			get_target_property(value ${target} ${property})
-			# remove compile flags that we inserted by
-			# target_precompiled_header
-			if(property STREQUAL "COMPILE_FLAGS")
-				string(REPLACE "${flags}" "" value "${value}")
-			endif()
-			if(NOT value STREQUAL "value-NOTFOUND")
-				set_target_properties(
-					"${pch_target}"
-					PROPERTIES
-					"${property}"
-					"${value}"
-					)
+				get_target_property(value ${target} ${property})
+				if(NOT value STREQUAL "value-NOTFOUND")
```

Comment:
Instead of testing for `-NOTFOUND` you could use `get_property` to check is that property is available.